src/lib/sql/sql_translator.cpp

```diff
@@ -355,11 +356,21 @@ std::shared_ptr<AbstractLQPNode> SQLTranslator::_translate_join(const hsql::Join
   auto left_node = _translate_table_ref(*join.left);
   auto right_node = _translate_table_ref(*join.right);
 
-  const hsql::Expr& condition = *join.condition;
+  std::vector<const hsql::Expr*> condition_list;
+  _flatten_conjunctive_expression_tree(join.condition, condition_list);
+
+  auto get_side =
```

Comment:
Consider it a matter of my taste, but I find this super ugly. Please just pass left_node, right_node into _get_sides_from_operator_expressions and call SQLTranslator::_get_side there.

Reply:
Fine, I'll try to combine with suggestions of @mrks on this matter.